prediction_market_agent/agents/utils.py

```diff
@@ -44,24 +45,36 @@ def microchain_task_from_market(
 {memories}
 """
 
+EXTRACT_REASONINGS_TEMPLATE = """
+You are an agent that trades in prediction markets. You have a collection of memories that record your
+actions, and your reasoning behind them.
+
+Below you can find a TWEET related to previous trades that you placed. Produce a summary of the reasonings, extracted from the memories, that can explain why the trade was placed.
+
+TWEET: $TWEET
+
+MEMORIES:
+{memories}
+"""
+
 
 def market_is_saturated(market: AgentMarket) -> bool:
     return market.current_p_yes > 0.95 or market.current_p_no > 0.95
 
 
-def memories_to_learnings(memories: t.Sequence[MemoryContainer], model: str) -> str:
-    """
-    Synthesize the memories into an intelligible summary that represents the
-    past learnings.
-    """
+def _summarize_learnings(
+    memories: t.Sequence[MemoryContainer],
+    prompt_template: PromptTemplate,
+    model: str = "gpt-4o-2024-05-13",
+) -> str:
     llm = ChatOpenAI(
```

Comment:
Consolidated the new method and the prev existing method into a single common method that gets called twice